src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentService.kt

```diff
@@ -14,54 +16,67 @@ import javax.transaction.Transactional
   val appointmentRepository: AppointmentRepository,
+  val communityAPIBookingService: CommunityAPIBookingService,
   val authUserRepository: AuthUserRepository,
 ) {
   fun createOrUpdateAppointment(
+    referral: Referral,
     appointment: Appointment?,
     durationInMinutes: Int,
     appointmentTime: OffsetDateTime,
+    appointmentType: AppointmentType,
     createdByUser: AuthUser
   ): Appointment {
-    val noAppointment = appointment == null
-    if (noAppointment) {
-      return createAppointment(durationInMinutes, appointmentTime, createdByUser)
+
+    val initialAppointmentRequired = appointment == null
+    if (initialAppointmentRequired) {
+      val deliusAppointmentId = communityAPIBookingService.book(referral, null, appointmentTime, durationInMinutes, appointmentType)
+      val createdAppointment = createAppointment(durationInMinutes, appointmentTime, deliusAppointmentId, createdByUser)
+      return appointmentRepository.save(createdAppointment)
     }
 
-    val appointmentWithoutAttendanceRecorded = appointment!!.attended == null
-    if (appointmentWithoutAttendanceRecorded) {
-      return updateAppointment(appointment, durationInMinutes, appointmentTime)
+    val updateCurrentAppointmentRequired = appointment!!.attended == null
+    if (updateCurrentAppointmentRequired) {
+      val deliusAppointmentId = communityAPIBookingService.book(referral, appointment, appointmentTime, durationInMinutes, appointmentType)
+      val updatedAppointment = updateAppointment(appointment, durationInMinutes, appointmentTime, deliusAppointmentId)
+      return appointmentRepository.save(updatedAppointment)
     }
 
-    val appointmentWithNonAttendance = appointment.attended == Attended.NO
-    if (appointmentWithNonAttendance) {
-      return createAppointment(durationInMinutes, appointmentTime, createdByUser)
+    val additionalAppointmentRequired = appointment.attended == Attended.NO
+    if (additionalAppointmentRequired) {
+      val deliusAppointmentId = communityAPIBookingService.book(referral, null, appointmentTime, durationInMinutes, appointmentType)
+      val additionalAppointment = createAppointment(durationInMinutes, appointmentTime, deliusAppointmentId, createdByUser)
+      return appointmentRepository.save(additionalAppointment)
     }
+
     throw IllegalStateException("Is it not possible to update an appointment that has already been attended")
   }
 
   private fun createAppointment(
     durationInMinutes: Int,
     appointmentTime: OffsetDateTime,
+    deliusAppointmentId: Long?,
     createdByUser: AuthUser,
   ): Appointment {
-    val appointment = Appointment(
+    return Appointment(
       id = UUID.randomUUID(),
       appointmentTime = appointmentTime,
       durationInMinutes = durationInMinutes,
+      deliusAppointmentId = deliusAppointmentId,
       createdBy = authUserRepository.save(createdByUser),
       createdAt = OffsetDateTime.now()
     )
-    appointmentRepository.save(appointment)
```

Comment:
Out of curiosity, what was the intent of moving `save` outside?

The function's verb "create"/"update" kind of implies it'd be "created", which I'd assume implies persistence

Reply:
Save was moved outside as the appointment needs to have the delius appointment id added to it before being saved

Reply:
is this still true? looks like we pass the delius id into this method. would be good to put the save back in here if possible

Reply:
Will move the save - you are correct